ui/pages/routes/routes.component.js

```diff
@@ -294,6 +304,23 @@ export default class Routes extends Component {
     }
   };
 
+  isTokenDetectionSupportedForCurrentNetwork = async () => {
+    const fetchedTokenData = await fetchWithCache(
+      `${TOKEN_API_METASWAP_CODEFI_URL}${this.props.providerChainId}`,
+    );
+
+    if (fetchedTokenData.error) {
+      return false;
+    }
+
+    return true;
+  };
```

Comment:
As discussed on Slack: https://consensys.slack.com/archives/C02GH833686/p1638979091138100?thread_ts=1638965448.130400&cid=C02GH833686.
This API is being used in order to check whether the current network supports token detection or not.
Using the `fetchTokenList` from the `TokenListController` changes the state and there is no return (side effect function). Also, `fetchFromDynamicTokenList` that is being called by `fetchTokenList` is currently possible only on mainnet.